asserts/model.go

```diff
@@ -602,9 +638,21 @@ func assembleModel(assert assertionBase) (Assertion, error) {
 	}
 
 	var modSnaps *modelSnaps
+	grade := ModelGradeUnset
 	if extended {
-		// TODO: support and consider grade!
-		modSnaps, err = checkExtendedSnaps(extendedSnaps, base)
+		gradeStr, err := checkOptionalString(assert.headers, "grade")
+		if err != nil {
+			return nil, err
+		}
+		if gradeStr != "" && !strutil.ListContains(validModelGrades, gradeStr) {
+			return nil, fmt.Errorf("grade for model must be stable|unstable")
+		}
+		grade = ModelStable
```

Comment:
This confused me for a second until I noticed this whole section is under `if extended {...}`. Otherwise I would have expected `ModelGradeUnset`. This is surely correct though.